kernel/src/table_changes/physical_to_logical.rs

```diff
@@ -0,0 +1,191 @@
+use std::collections::HashMap;
+use std::iter;
+
+use itertools::Itertools;
+use url::Url;
+
+use crate::actions::deletion_vector::split_vector;
+use crate::expressions::{column_expr, Scalar};
+use crate::scan::state::GlobalScanState;
+use crate::scan::{parse_partition_value, ColumnType, ScanResult};
+use crate::schema::{ColumnName, DataType, SchemaRef, StructField, StructType};
+use crate::{DeltaResult, Engine, Error, Expression, ExpressionRef, FileMeta};
+
+use super::resolve_dvs::ResolvedCdfScanFile;
+use super::scan_file::{CdfScanFile, CdfScanFileType};
+
+/// Returns a map from change data feed column name to an expression that generates the row data.
+#[allow(unused)]
+fn get_generated_columns(scan_file: &CdfScanFile) -> DeltaResult<HashMap<&str, Expression>> {
```

Comment:
reason for `&str` over `ColumnName`? i suppose cheaper/no alloc?

Reply:
also note we _might_ want to rename this (or at least just be aware of this naming in the future) since there is an actual table feature called generated columns

Reply:
The values are fetched with `field_name: &String` from `ColumnType::Selected(field_name)`. Doesn't make sense to convert to ColumnName only to look at the hashmap.
```rust
ColumnType::Selected(field_name /*: &String */) => {
    let generated_column = cdf_columns.remove(field_name.as_str());
    ...
}
```

Reply:
Yeah these are top-level columns.